### ISSUE TITLE
is gaussian elimination on?

### ISSUE THREAD

**Issue by edredre** [msoos/cryptominisat#452]
Hi, I'm not sure whether the Gaussian elimination is on in my installation of cryptominisat. Is there a way to understand if it is on or not (apart from installing as suggested in the main page of this website)? Thanks!

**Comment by msoos** [comment]
Hi!

In the README.md (also on the main github page) you will find the
instructions how to turn it on  :) It needs to be recompiled with the right
flags and then ran with the correct flags. Please check it out and let me
know if it's not clear or if any other issue comes up :)

Good luck!

**Comment by edredre** [comment]
Hi Mate, I recompiled as specified but run without any flag. Which flag do I have to specify? I cannot find any indication on this in the main github page...

**Comment by msoos** [comment]
Sorry, on mobile :) Have you tired playing with the options explained in
the README? Have you tried setting to 0 the "--autodisablegauss"? What is
the output? Does it mention use and effectiveness of Gauss?

**Comment by edredre** [comment]
this is how I invoke it now (within a python program):

  subprocess.call('/usr/local/bin/cryptominisat5 --autodisablegauss 0 workingfile.cnf > outfile.txt' , shell=True)

and this is what happens:

ERROR: Some option you gave was wrong. Please give '--help' to get help
       Unkown option: unrecognised option '--autodisablegauss'

**Comment by edredre** [comment]
It seems that Gauss elimination has not been installed. Look at (the beginning of) this call:

MacBook-Air-di-Daniele:approxmc-py-4 Daniele$ cryptominisat5 binsearch.32.cnf
c Outputting solution to console
c CryptoMiniSat version 5.0.1
c CryptoMiniSat SHA revision GIT-notfound
c CryptoMiniSat compilation env CMAKE_CXX_COMPILER = /Library/Developer/CommandLineTools/usr/bin/c++ | CMAKE_CXX_FLAGS =  -std=c++11 -g -pthread -Wall -Wextra -Wunused -pedantic -Wsign-compare -fno-omit-frame-pointer -Wtype-limits -Wuninitialized -Wno-deprecated -Wstrict-aliasing -Wpointer-arith -Wheader-guard -Wpointer-arith -Wformat-nonliteral -Winit-self -Wparentheses -Wunreachable-code -ggdb3 -fPIC | COMPILE_DEFINES =  -DUSE_PTHREADS -DUSE_ZLIB | STATICCOMPILE = OFF | ONLY_SIMPLE = OFF | Boost_FOUND = 1 | TBB_FOUND =  | STATS = OFF | MYSQL_FOUND =  | SQLITE3_FOUND =  | ZLIB_FOUND = TRUE | VALGRIND_FOUND = FALSE | ENABLE_TESTING = OFF | M4RI_FOUND = FALSE | SLOW_DEBUG = OFF | ENABLE_ASSERTIONS = ON | PYTHON_EXECUTABLE = /usr/bin/python2.7 | PYTHON_LIBRARY = /usr/lib/libpython2.7.dylib | PYTHON_INCLUDE_DIRS = /usr/include/python2.7 | MY_TARGETS =  | compilation date time = Nov 30 2017 20:16:13
c compiled with gcc version 4.2.1 Compatible Apple LLVM 6.0 (clang-600.0.57)
c Executed with command line: cryptominisat5 binsearch.32.cnf
c Reading file 'binsearch.32.cnf'
c -- header says num vars:           4473
c -- header says num clauses:       14011

**Comment by edredre** [comment]
However I have (re)installed on my Mac cryptominisat with the options you specified. In particular, from the (already existing) folder 'build', I executed 

cmake -DUSE_GAUSS=ON ..
make

**Comment by edredre** [comment]
I have also tried to delete the previous folder build and perform 

mkdir build && cd build
cmake -DUSE_GAUSS=ON ..
make

but the situation doesn't change. I still have

COMPILE_DEFINES =  -DUSE_PTHREADS -DUSE_ZLIB

and

M4RI_FOUND = FALSE

whereas a colleague that has Gauss elimination working has

COMPILE_DEFINES =
-DUSE_GAUSS -DUSE_PTHREADS -DBOOST_TEST_DYN_LINK -DUSE_ZLIB -DUSE_M4RI

and

M4RI_FOUND = TRUE

Any help??

**Comment by msoos** [comment]
You need to install it too :) Just doing "make" will just re-build it. The binary in the folder with probably have gauss on. Have you tried running "./cryptominisat " from there to check? You need to install that cryptominisat. Do "sudo make install" after the "make" :)

**Comment by a1880** [comment]
`M4RI_FOUND` will be `FALSE` unless [m4ri ](https://bitbucket.org/malb/m4ri) is actually installed. What is the performance impact of leaving `m4ri` uninstalled?

**Comment by msoos** [comment]
Let's skip M4RI for the moment, and get gauss inside cryptominisat working, please. No need to mix stuff for the moment. Please, do "sudo make install" as I mentioned previously. Then the option should work and then you'll have it working as intended.

**Comment by edredre** [comment]
ok, now Gauss elimination is on! What about M4RI? Do I have to install it to improve performances? In case, how can I install it?

**Comment by edredre** [comment]
Another difference w.r.t. my colleague is that he also has DBOOST_TEST_DYN_LINK in the COMPILE_DEFINES... Does it have impact on performances too? How can I install?

I'm asking because we're running the same code on very similar machines (i7 1.7GHz and i7 2.8GHz, but my runs are much longer than his)...

Thanks a lot!!

**Comment by msoos** [comment]
DBOOST_TEST_DYN_LINK -- it's not used, only used in case you want to perform testing.

M4RI -- yes, that might be an issue. Recomplie with M4RI installed, and "sudo make install" again.

Have you installed as per the README? It states:

`sudo apt-get install libzip-dev libboost-program-options-dev libm4ri-dev libsqlite3-dev`

Please follow the instructions.

**Comment by a1880** [comment]
m4ri installation might be different under Mac OS.

**Comment by edredre** [comment]
Indeed, first in MacOS the command apt-get doesn't work. Ok, I've replaced the previous call with 

brew install libzip-dev libboost-program-options-dev libm4ri-dev libsqlite3-dev

but this is the output:

MacBook-Air-di-Daniele:approxmc-py-4 Daniele$ brew install libzip-dev libboost-program-options-dev libm4ri-dev libsqlite3-dev
Updating Homebrew...
==> Auto-updated Homebrew!
Updated 1 tap (homebrew/core).
==> New Formulae
mariadb-connector-odbc
==> Updated Formulae
amber                  clamav                 dscanner               guile@2.0              jenkins-job-builder    libtensorflow          oniguruma              vault
arangodb               crystal-icr            ecl                    hadolint               jhiccup                lynis                  picard-tools           w3m
autogen                crystal-lang           fn                     hg-fast-export         joplin                 lysp                   s-search               xtensor
bdw-gc                 dep                    gitbucket              htslib                 kite                   mapnik                 sagittarius-scheme     youtube-dl
bork                   diamond                godep                  ice                    kobalt                 mp3gain                samtools               zile
certbot                dnscrypt-proxy         guile                  iron-functions         libphonenumber         neko                   spigot
==> Deleted Formulae
mg3a                                                           mvptree                                                        talk-filters

Error: No available formula with the name "libzip-dev" 
==> Searching for a previously deleted formula (in the last month)...
Warning: homebrew/core is shallow clone. To get complete history run:
  git -C "$(brew --repo homebrew/core)" fetch --unshallow

Error: No previously deleted formula found.
==> Searching for similarly named formulae...
==> Searching local taps...
Error: No similarly named formulae found.
==> Searching taps...
==> Searching taps on GitHub...
Error: No formulae found in taps.
